mmf/datasets/builders/localized_narratives/database.py

```diff
@@ -55,18 +56,42 @@ def load_annotation_db(self, path):
             for line in f:
                 annotation = json.loads(line)
                 loc_narr = LocalizedNarrative(**annotation)
-                data.append(
-                    {
-                        "dataset_id": loc_narr.dataset_id,
-                        "image_id": loc_narr.image_id,
-                        "caption": loc_narr.caption,
-                        "feature_path": self._feature_path(
-                            loc_narr.dataset_id, loc_narr.image_id
-                        ),
-                    }
-                )
+                data_detail = {
+                    "image_id": loc_narr.image_id,
+                    "feature_path": self._feature_path(
+                        loc_narr.dataset_id, loc_narr.image_id
+                    ),
+                }
+                if (
+                    hasattr(self.config, "trace_mode")
+                    and self.config.trace_mode is not None
+                ):
+                    data_detail["utterances"] = self._utterances(loc_narr.timed_caption)
+                    data_detail["utterance_times"] = self._utterance_times(
+                        loc_narr.timed_caption
+                    )
+                    data_detail["traces"] = self._traces(loc_narr.traces)
+                else:
+                    data_detail["caption"] = loc_narr.caption
+
+                data.append(data_detail)
+
         self.data = data
 
+    def _utterances(self, timed_caption):
+        return [caption["utterance"] for caption in timed_caption]
+
+    def _utterance_times(self, timed_caption):
+        times = [
+            [caption["start_time"], caption["end_time"]] for caption in timed_caption
+        ]
+        return torch.FloatTensor(times)
```

Comment:
Preferred format is:
```suggestion
        return torch.tensor(times, dtype=torch.float)
```